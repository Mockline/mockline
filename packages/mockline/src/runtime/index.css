@import "tailwindcss";
@import "assets/css/animation.css";

@variant dark (&:where(.dark, .dark *));

@plugin "@tailwindcss/typography";

@theme {
  --color-neutral: var(--ui-bg);
  --color-neutral-elevated: var(--ui-bg-elevated);
  --color-border: var(--ui-border);
}

@layer base {
  :root {
    color-scheme: light;

    --ui-text-dimmed: var(--ui-color-neutral-400);
    --ui-text-muted: var(--ui-color-neutral-500);
    --ui-text-toned: var(--ui-color-neutral-600);
    --ui-text: var(--ui-color-neutral-700);
    --ui-text-highlighted: var(--ui-color-neutral-900);

    --ui-bg: var(--color-white);
    --ui-bg-elevated: var(--ui-color-neutral-100);
    --ui-bg-accented: var(--ui-color-neutral-200);
    --ui-bg-inverted: var(--ui-color-neutral-900);

    --ui-border: var(--ui-color-neutral-200);
    --ui-border-accented: var(--ui-color-neutral-300);
    --ui-border-inverted: var(--ui-color-neutral-900);

    --ui-radius: var(--radius);

    /*--ui-slate: var(--color-slate-500);
    --ui-gray: var(--color-gray-500);
    --ui-zinc: var(--color-zinc-500);
    --ui-stone: var(--color-stone-500);
    --ui-neutral: var(--color-neutral-500);
    --ui-red: var(--color-red-500);
    --ui-orange: var(--color-orange-500);
    --ui-amber: var(--color-amber-500);
    --ui-yellow: var(--color-yellow-500);
    --ui-lime: var(--color-lime-500);
    --ui-green: var(--color-green-500);
    --ui-emerald: var(--color-emerald-500);
    --ui-teal: var(--color-teal-500);
    --ui-cyan: var(--color-cyan-500);
    --ui-sky: var(--color-sky-500);
    --ui-blue: var(--color-blue-500);
    --ui-indigo: var(--color-indigo-500);
    --ui-violet: var(--color-violet-500);
    --ui-purple: var(--color-purple-500);
    --ui-fuchsia: var(--color-fuchsia-500);
    --ui-pink: var(--color-pink-500);
    --ui-rose: var(--color-rose-500);*/
  }

  .dark {
    color-scheme: dark;

    --ui-text-dimmed: var(--ui-color-neutral-500);
    --ui-text-muted: var(--ui-color-neutral-400);
    --ui-text-toned: var(--ui-color-neutral-300);
    --ui-text: var(--ui-color-neutral-200);
    --ui-text-highlighted: var(--color-white);

    --ui-bg: var(--ui-color-neutral-900);
    --ui-bg-elevated: var(--ui-color-neutral-800);
    --ui-bg-accented: var(--ui-color-neutral-700);
    --ui-bg-inverted: var(--color-white);

    --ui-border: var(--ui-color-neutral-800);
    --ui-border-accented: var(--ui-color-neutral-700);
    --ui-border-inverted: var(--color-white);
  }
}

