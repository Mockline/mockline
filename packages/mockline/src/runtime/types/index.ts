import type { Color } from '../utils/colors'

export * from '../components/elements/Button.vue'

export type ModuleOptions = {
  /**
   * Prefix for all components
   */
  prefix?: string,
  /**
   * Enable Nuxt Content
   * @default false
   */
  content?: boolean
  /**
   * The colors to use
   */
  colors?: Color[]
  /**
   * Enable Color Transitions
   * @default false
   */
  transitions?: boolean
}
