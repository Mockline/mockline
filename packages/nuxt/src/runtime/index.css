@import "@mockline/themes";
@import "./mockline.css";
