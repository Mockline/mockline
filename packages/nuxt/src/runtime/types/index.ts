// Elements
export * from '../components/Button.vue'
export * from '../components/NavMenu.vue'

// Layout
export * from '../components/App.vue'
export * from '../components/Aside.vue'
export * from '../components/Card.vue'
export * from '../components/Header.vue'
export * from '../components/Main.vue'
export * from '../components/Page.vue'
export * from '../components/PageBody.vue'
export * from '../components/PageHeader.vue'

// Settings
export * from '../components/ThemeToggle.vue'

// Overlays
export * from '../components/Toasts.vue'

// Content
export * from '../components/content/ContentNavigationTree.vue'
export * from '../components/content/ContentSurround.vue'
export * from '../components/content/ContentToc.vue'
export * from '../components/content/ContentTocLinks.vue'

// Global
export * from '../components/prose/Callout.vue'
export * from '../components/prose/CodeGroup.vue'
export * from '../components/prose/ProseCode.vue'
export * from '../components/prose/CodeButton.vue'
