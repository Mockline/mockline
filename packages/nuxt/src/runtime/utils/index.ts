export function transformSlot(slot: any, index: number): any {
  if (typeof slot.type === 'symbol') {
    return slot.children?.map(transformSlot)
  }
  return {
    label: slot.props?.filename || slot.props?.label || `${index}`,
    icon: slot.props?.icon,
    component: slot
  }
}

export * from './useComponent'
