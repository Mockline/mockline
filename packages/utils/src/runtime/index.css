@import "tailwindcss";

@variant dark (&:where(.dark, .dark *));

@plugin "@tailwindcss/typography";
