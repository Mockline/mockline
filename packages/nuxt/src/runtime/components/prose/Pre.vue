<script setup lang="ts">
import { computed, ref } from 'vue'
import { useClipboard } from '@vueuse/core'
import { toast } from 'vue-sonner'
import type { ProsePreProps } from '@mockline/themes'
import CodeIcon from './CodeIcon.vue'
import appConfig from '#build/app.config'
import { useComponent } from '#mockline/utils/useComponent'

const props = defineProps<ProsePreProps>()

const clipboard = useClipboard()

const copied = ref(false)

function copy(): void {
  clipboard.copy(props.code || '')

  copied.value = true

  toast.success('Copied to clipboard')

  setTimeout(() => {
    copied.value = false
  }, 2000)
}

const componentProps = computed(() => {
  return {
    ...props,
    filename: !!props.filename,
  }
})

const { getClasses } = useComponent('prosePre', componentProps)
</script>

<template>
  <div :class="getClasses('root')">
    <div
      v-if="filename && !hideHeader"
      :class="getClasses('header')"
    >
      <CodeIcon :icon :filename :class="getClasses('icon')" />

      <span :class="getClasses('filename')">{{ filename }}</span>
    </div>

    <MButton
      :icon="copied ? appConfig.mockline.icons.copyCheck : appConfig.mockline.icons.copy"
      :color="copied ? 'primary' : 'neutral'"
      variant="outline"
      size="sm"
      aria-label="Copy code to clipboard"
      :class="getClasses('copy')"
      tabindex="-1"
      @click="copy"
    />

    <pre :class="getClasses('base', props.class)" v-bind="$attrs"><slot /></pre>
  </div>
</template>

<style>
.shiki span.line{display:block}.shiki span.line.highlight{margin:0 -16px;padding:0 16px;@apply bg-[var(--ui-bg-accented)]/50}
</style>
