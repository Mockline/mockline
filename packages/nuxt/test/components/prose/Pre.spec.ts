import { describe, it, expect } from 'vitest'
import { renderSuspended } from '@nuxt/test-utils/runtime'
import Pre from '../../src/runtime/components/prose/Pre.vue'

describe('Pre', () => {
  it('can mount component', async () => {
    const html = await renderSuspended(Pre)
    expect(html.html()).toMatchSnapshot()
  })
})
