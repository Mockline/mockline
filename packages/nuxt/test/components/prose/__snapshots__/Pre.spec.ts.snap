// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Pre > can mount component 1`] = `
"<div id="test-wrapper">
  <div class="relative my-5 group not-[.code-group_&amp;]:bg-[var(--color-bg-primary)] not-[.code-group_&amp;]:border not-[.code-group_&amp;]:border-[var(--color-border-primary)] not-[.code-group_&amp;]:rounded-base not-[.code-group_&amp;]:p-2 dark">
    <!--v-if--><button class="inline-flex items-center justify-center font-medium cursor-pointer select-none focus-visible:outline-none touch-none tap-highlight-none rounded-[var(--radius-base)] text-xs gap-1.5 text-[var(--color-fg-primary)] hover:bg-[var(--color-bg-quaternary)] active:bg-[var(--color-bg-quaternary)] focus-visible:ring-2 focus-visible:ring-[var(--color-fg-primary)]/50 focus-visible:ring-inset disabled:hover:bg-transparent p-1.5 duration-200 ease-in-out active:translate-y-px absolute top-[11px] right-[11px] opacity-0 group-hover:opacity-100 transition" aria-label="Copy code to clipboard" tabindex="-1"><span class="iconify i-lucide:copy shrink-0 relative size-4" aria-hidden="true"></span>
      <!--v-if-->
      <!--v-if-->
    </button><pre class="group font-mono text-sm/6 p-1 rounded-base whitespace-pre-wrap break-words overflow-x-auto"></pre>
  </div>
</div>"
`;
